dali/operators/video/frames_decoder_gpu.cc

```diff
@@ -465,12 +465,12 @@ int FramesDecoderGpu::HandlePictureDisplay(CUVIDPARSERDISPINFO *picture_display_
   videoProcessingParameters.unpaired_field = 0;
   videoProcessingParameters.output_stream = stream_;
 
-  if (current_pts_ == AV_NOPTS_VALUE) {
-    // if != AV_NOPTS_VALUE, it's not the first frame from the packet
-    // first frame from this packet, pop it from the queue and remember it.
+  // If there are no piped pts, use the last pts available (unexpected extra frame)
+  if (!piped_pts_.empty()) {
```

Comment:
I wonder what happens if we have 2 packets in the processing, and the first will produce two frames and the last only one. We will affiliate the first frame with first packet, and the second and third with the second packet.